pilot/pkg/bootstrap/server.go

```diff
@@ -273,6 +276,17 @@ func NewServer(args *PilotArgs) (*Server, error) {
 		return nil, fmt.Errorf("config validation: %v", err)
 	}
 
+	if err := s.initDNSListener(args); err != nil {
+		return nil, fmt.Errorf("dns listener: %v", err)
+	}
+
+	// Respond to CoreDNS gRPC queries.
+	s.addStartFunc(func(stop <-chan struct{}) error {
+		dnsSvc := dns.InitCoreDNS(s.secureGRPCServerDNS, "", s.environment.IstioConfigStore)
```

Comment:
are we using the same underlying `secureGRPCServerDNS` server to register different services and listen over different ports?

Does that cause any issues? I have not re-used grpc server object like that before.

Reply:
No, we have 2 gRPC servers, one secure and one not secure. 

One each we have different services. I fixed the code to also register on the unsecure grpc server.

We already have multiple services - CA and xDS (and will have MCP)